packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -294,7 +294,7 @@ class _DevFSHttpWriter {
     } catch (e) {
       if (retry < kMaxRetries) {
         printTrace('Retrying writing "$deviceUri" to DevFS due to error: $e');
-        _scheduleWrite(deviceUri, content, progressReporter, retry + 1);
+        await _scheduleWrite(deviceUri, content, progressReporter, retry + 1);
```

Comment:
Note that the progressReporter code is dead now. It was only used by the loading screen.